src/main/java/io/javarig/generator/TypeBasedGenerator.java

```diff
@@ -2,8 +2,12 @@
 
 import java.lang.reflect.Type;
 
+/**
+ * type generator for classes that require an extra type
```

Comment:
This is not clear 
I think better say :  type generator for classes that require the Type object that holds information about the object to generate